server/test/integration/dbTest/getHistoricAlertsByAlertApiKeyTest.js

```diff
@@ -0,0 +1,296 @@
+const SessionState = require('../../../SessionState.js')
+
+describe('db.js integration tests: getHistoricAlertsByAlertApiKey', () => {
+  describe('if there are no installations with the given Alert API Key', () => {
+    beforeEach(async () => {
+      await db.clearSessions()
+      await db.clearButtons()
+      await db.clearInstallations()
+
+      // Insert a single installation with a single button that has a single session that doesn't match the Alert API Key that we ask for
+      await db.createInstallation('name', 'responderNumber', '{"fallbackNumber"}', '{"cat1"}', 'alertApiKey')
+      const installationId = (await db.getInstallations())[0].id
+      const buttonId = '51b8be5678bf5ade9bf6a5958b2a4a45'
+      await db.createButton(buttonId, installationId, 'unit', 'phoneNumber', 'serialNumber')
+      await db.createSession(installationId, buttonId, 'unit', 'phoneNumber', 5, 95, new Date('2021-01-20T06:20:19.000Z'))
+    })
+
+    afterEach(async () => {
+      await db.clearSessions()
+      await db.clearButtons()
+      await db.clearInstallations()
+    })
+
+    it('should return an empty array', async () => {
+      const rows = await db.getHistoricAlertsByAlertApiKey('not alertApiKey', 10, 50000)
+
+      expect(rows).to.eql([])
+    })
+  })
+
+  describe('if there are no sessions for the installation with the given Alert API Key', () => {
+    beforeEach(async () => {
+      await db.clearSessions()
+      await db.clearButtons()
+      await db.clearInstallations()
+
+      // Insert a single installation with a single button that has a single session that doesn't match the Alert API Key that we ask for
+      await db.createInstallation('name', 'responderNumber', '{"fallbackNumber"}', '{"cat1"}', 'not our API key')
+      const installationId = (await db.getInstallations())[0].id
+      const buttonId = '51b8be5678bf5ade9bf6a5958b2a4a45'
+      await db.createButton(buttonId, installationId, 'unit', 'phoneNumber', 'serialNumber')
+      await db.createSession(installationId, buttonId, 'unit', 'phoneNumber', 5, 95, new Date('2021-01-20T06:20:19.000Z'))
+
+      // Insert a single installation with no sessions that matches the Alert API Key that we ask for
+      this.alertApiKey = 'alertApiKey'
+      await db.createInstallation('name', 'responderNumber', '{"fallbackNumber"}', '{"cat1"}', this.alertApiKey)
+    })
+
+    afterEach(async () => {
+      await db.clearSessions()
+      await db.clearButtons()
+      await db.clearInstallations()
+    })
+
+    it('should return an empty array', async () => {
+      const rows = await db.getHistoricAlertsByAlertApiKey(this.alertApiKey, 10, 50000)
+
+      expect(rows).to.eql([])
+    })
+  })
+
+  describe('if there is one matching session', () => {
+    beforeEach(async () => {
+      await db.clearSessions()
+      await db.clearButtons()
+      await db.clearInstallations()
+
+      // Insert a single installation with a two buttons and more than maxHistoricAlerts sessions
```

Comment:
where does `maxHistoricAlerts` get set? it looks like only one session is created here - this comment doesn't seem to match that?

Reply:
You're right. Comment is copied from elsewhere and isn't correct. I'll fix it.

Reply:
But to answer question about `maxHistoricAlerts`, it's sent as a parameter to `getHistoricAlertsByAlertApiKey` in the test.

Reply:
ah yes, I see that. thanks!

Reply:
Fixed.